react/features/invite/components/AddPeopleDialog.web.js

```diff
@@ -0,0 +1,260 @@
+            && this._multiselect) {
+            this._multiselect.clear();
+        }
+    }
+
+    /**
+     * Renders the content of this component.
+     *
+     * @returns {XML}
+     */
+    render() {
+        return (
+            <Dialog
+                okDisabled = { this._isAddDisabled() }
+                okTitleKey = 'addPeople.add'
+                onSubmit = { this._onSubmit }
+                titleKey = 'addPeople.title'
+                width = 'small'>
+                { this._getUserInputForm() }
+            </Dialog>
+        );
+    }
+
+    /**
+     * Renders the input form.
+     *
+     * @returns {XML}
+     * @private
+     */
+    _getUserInputForm() {
+        const { t } = this.props;
+
+        return (
+            <div className = 'add-people-form-wrap'>
+                <MultiSelectAutocomplete
+                    isDisabled
+                        = { this.state.addToCallInProgress || false }
+                    noMatchesFound = { t('addPeople.noResults') }
+                    onSelectionChange = { this._onSelectionChange }
+                    placeholder = { t('addPeople.searchPlaceholder') }
+                    ref = { this._setMultiSelectElement }
+                    resourceClient = { this._resourceClient }
+                    shouldFitContainer = { true }
+                    shouldFocus = { true } />
+            </div>
+        );
+    }
+
+    /**
+     * Indicates if the Add button should be disabled.
+     *
+     * @returns {boolean} - True to indicate that the Add button should
+     * be disabled, false otherwise.
+     * @private
+     */
+    _isAddDisabled() {
+        return !this.state.inviteItems.length
+            || this.state.addToCallInProgress;
+    }
+
+    /**
+     * Handles a selection change.
+     *
+     * @param {Map} selectedItems - The list of selected items.
+     * @private
+     * @returns {void}
+     */
+    _onSelectionChange(selectedItems) {
+        const selectedIds = selectedItems.map(o => o.item);
+
+        this.setState({
+            inviteItems: selectedIds
+        });
+    }
+
+    /**
+     * Handles the submit button action.
+     *
+     * @param {Object} e - The event notifying us for the submit.
+     * @private
+     * @returns {void}
+     */
+    _onSubmit() {
+        if (!this._isAddDisabled()) {
+            this.setState({
+                addToCallInProgress: true
+            });
+
+            invitePeople(
+                this.props._inviteServiceUrl,
+                this.props._inviteUrl,
+                this.props._jwt,
+                this.state.inviteItems)
+            .then(() => {
+                this.setState({
+                    addToCallInProgress: false
+                });
+            })
+            .catch(() => {
+                this.setState({
+                    addToCallInProgress: false,
+                    addToCallError: true
+                });
+            });
```

Comment:
fyi, a benign and not-worth-fixing error about setting state on an unmounted component is printed here by react if the modal is closed before invitePeople returns.